veracruz-server-test/src/main.rs

```diff
@@ -843,6 +842,63 @@ mod tests {
         server_thread: JoinHandle<Result<(), String>>,
     }
 
+    struct Buffers {
+        // Read buffer used by mbedtls for cyphertext.
+        read_buffer: Vec<u8>,
+        // Write buffer used by mbedtls for cyphertext.
+        write_buffer: Option<Vec<u8>>,
+    }
+
+    struct InsecureConnection {
```

Comment:
Is this a duplicate of the previous `InsecureConnection` declaration? Is it necessary to duplicate it here?

Reply:
I think it's a separate thing, really.

Reply:
Would it make sense to rename it to prevent confusion?

Reply:
Each place where I use `mbedtls` there's a similar set-up with `Buffers` and `InsecureConnection` for interfacing with `mbedtls`, but they're not quite identical and they're file-local.